### DISCUSSION TITLE
What is the best way to update the OLM itself?

### DISCUSSION THREAD

**Discussion by MarcManz-1** [operator-framework/operator-lifecycle-manager#2974] [Q&A]
As long as there is no documentation available on how to update the OLM, can somebody please explain here what is the procedure to update the OLM?

Currently, I run the OLM in my test cluster to check out how to operate it. One part of this test is to update the OLM to a newer version. My current procedure is to uninstall the OLM via the operater-sdk and install it in a newer version. 
But during my test, either the installation or the uninstallation failed several times. 
If you would like to update the OLM in a production cluster, there is always uncertainty when uninstalling it, and all installed operators are removed.  
Reinstalling the OLM also requires reapplying every single operator subscription.

**Comment by coarasa** [comment]
A proper lifecycle is certainly needed for OLM itself
